modules/fbs/includes/fbs.loan.inc

```diff
@@ -120,22 +151,37 @@ function fbs_loan_renew($account, $loans) {
         $loan->loanDetails->loanType == 'interLibraryLoan' ?
         DingProviderLoan::STATUS_RENEWAL_REQUESTED :
         DingProviderLoan::STATUS_RENEWED;
+
+      // This loan was succefully renewed. Remember this for the rest of the
+      // session.
+      $renewed_ids[] = $loan->loanDetails->loanId;
     }
     else {
-      $result[$loan->loanDetails->loanId] = DingProviderLoan::STATUS_NOT_RENEWED;
-      if (in_array('deniedMaxRenewalsReached', $loan->renewalStatus)) {
-        $result[$loan->loanDetails->loanId] = DingProviderLoan::STATUS_RENEWAL_NUM_EXCEEDED;
-      }
-      if (in_array('deniedReserved', $loan->renewalStatus)) {
-        $result[$loan->loanDetails->loanId] = DingProviderLoan::STATUS_RENEWAL_RESERVED;
-      }
+      $result[$loan->loanDetails->loanId] = _fbs_loan_get_renewal_status($loan);
     }
   }
 
   // Clear ding session cache.
   if (module_exists('ding_session_cache')) {
     ding_session_cache_clear('fbs', 'loans');
+    ding_session_cache_set('fbs', 'renewed_ids', $renewed_ids);
   }
 
   return $result;
 }
+
+/**
+ * Helper function to extract renewalStatus from the FBS loan and convert it to
+ * to Ding provider renewal status constants.
+ *
+ * Used when renewing and showing loan list.
```

Comment:
Note the Scrutinizer feedback.

Remember to provide a full DocBlock including parameter and return types.

I do not think where a function is used is very helpful. It quickly gets obsolete and most editors will help you find it pretty quickly.

Reply:
I'll provide a full DocBlock and remove the "used when" comment.